src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -357,12 +363,7 @@ public CloudSdk build() {
 
       // Default SDK path
       if (sdkPath == null) {
-        Path discoveredSdkPath = new PathResolver().getCloudSdkPath();
-        if (discoveredSdkPath == null) {
-          throw new AppEngineException("Google Cloud SDK path was not provided and could not be"
-              + " found in any known install locations.");
-        }
-        sdkPath = discoveredSdkPath;
+        sdkPath = discoverSdkPath();
```

Comment:
Seems a little overly complex when you can just specify the location of your sdk in your tests directly and appears to have some very osgi specific work arounds. Is there a reason this is absolutely required?